build_tools/publisher/gem_publisher.rb

```diff
@@ -0,0 +1,40 @@
+require 'govuk_template/version'
+require 'tmpdir'
+require 'open3'
+
+module Publisher
+  class GemPublisher
+    attr_accessor :version
+
+    GIT_URL = "https://#{ENV['GITHUB_TOKEN']}@github.com/alphagov/govuk_template.git"
+
+    def initialize(version = GovukTemplate::VERSION)
+      @version = version
+    end
+
+    def publish
+      puts "Pushing govuk_template-#{GovukTemplate::VERSION}"
+      run "gem push pkg/govuk_template-#{GovukTemplate::VERSION}.gem"
+      Dir.mktmpdir("govuk_template_gem") do |dir|
+        run "git clone -q #{GIT_URL.shellescape} #{dir.shellescape}"
```

Comment:
Excellent catch 👏

I've run a local test of this, simulating a bad connection to the repository domain and it sends this to `stderr`:

```
fatal: unable to access 'https://*ENV*@github.com/alphagov/govuk_template.git/': Failed to connect to github.com port 443: Connection refused
```
(where \*ENV\* is the github token.)

That being the case, I've rewritten how we run shell commands in 146aa55, to allow custom errors and used them to mask this kind of output.